R/tax_range_geo.R

```diff
@@ -0,0 +1,251 @@
+#' \code{\link[geosphere:areaPolygon]{geosphere::areaPolygon()}}.
+#' - Latitudinal: the "lat" method calculates the palaeolatitudinal
+#' range of a taxa. It does so for each taxa in `occdf` by finding their most
+#' northern and southern occurrence. The palaeolatitudinal range of each taxa
+#' is also calculated.
+#' - Maximum Great Circle Distance: the "gcd" method calculates the maximum
+#' Great Circle Distance between occurrences for each taxa in `occdf`. It does
+#' so using \code{\link[fields:rdist.earth]{fields::rdist.earth()}}. This
+#' function calculates Great Circle Distance using the Haversine method with
+#' the radius of the Earth set to the 6378.388 km (equatorial radius).
+#' - Occupied cells: the "occ" method calculates the number of occupied
+#' grid cells. It does so using discrete hexagonal grids via the
+#' \code{\link[h3jsr]{h3jsr}} package. This package relies on Uber's H3 library,
+#' a geospatial indexing system that partitions the world into hexagonal cells:
+#' \url{https://h3geo.org/docs}. In H3, 16 different resolutions are available:
+#' \url{https://h3geo.org/docs/core-library/restable}. In the implementation of
+#' the `tax_range_geo()` function, the resolution is defined by the user-input
+#' `spacing` which represents the distance between the centroid of adjacent
+#' cells. Using this distance, the function identifies which resolution is most
+#' similar to the input `spacing`, and uses this resolution.
+#'
+#' @section Developer(s):
+#' Lewis A. Jones
+#' @section Reviewer(s):
+#' To be reviewed
+#' @importFrom geosphere areaPolygon
+#' @importFrom grDevices chull
+#' @importFrom fields rdist.earth
+#' @importFrom h3jsr point_to_h3
+#' @examples
+#' # Grab internal data and set-up
+#' occdf <- tetrapods
+#' # Convex hull
+#' tax_range_geo(occdf = occdf, name = "accepted_name", method = "con")
+#' # Latitudinal range
+#' tax_range_geo(occdf = occdf, name = "accepted_name", method = "lat")
+#' # Great Circle Distance
+#' tax_range_geo(occdf = occdf, name = "accepted_name", method = "gcd")
+#' # Occupied grid cells
+#' tax_range_geo(occdf = occdf, name = "accepted_name", method = "occ")
+#' @export
+tax_range_geo <- function(occdf,
+                          name = "name",
+                          lng = "lng",
+                          lat = "lat",
+                          method = "lat",
+                          spacing = 100) {
+
+  #=== Handling errors ===
+  if (!is.data.frame(occdf)) {
+    stop("`occdf` should be a dataframe")
+  }
+
+  if (any(c(name, lng, lat) %in% colnames(occdf) == FALSE)) {
+    stop("Either `name`, `lng`, or `lat`, do not exist in `occdf`")
+  }
+
+  if (!is.character(method)) {
+    stop("`method` is not of character class")
+  }
+
+  if (!is.numeric(occdf[,lat]) || !is.numeric(occdf[,lng])) {
+    stop("`lng` and/or `lat` columns are not of numeric class")
+  }
+
+  if (any(is.na(occdf[,name]))) {
+    stop("The `name` column contains NA values")
```

Comment:
It should maybe be mentioned somewhere in the documentation that these should be filtered out beforehand.